swfiles/sw_fitpowder.m

```diff
@@ -592,6 +695,25 @@ function fix_parameters(obj, iparams)
                obj.bounds(iparam, :) = obj.params(iparam);
             end
         end
+        function bg = find_indices_and_mean_of_bg_bins(obj)
+             % start with a seed of indices of likely non-signal bins
+            iseed = find(isfinite(obj.y) & obj.y < mean(obj.y(:), 'omitnan'));
+            [ysort, isort] = sort(obj.y(iseed), 'descend');
+            bg = ysort(isort(int32(numel(isort)/2))); % default to median
+            prev_skew = inf;
+            for ipt = 1:numel(isort)
+                this_mean = mean(ysort(ipt:end));
+                this_skew = mean((ysort(ipt:end) - this_mean).^3);
+                if this_skew < 0 || this_skew > prev_skew
+                    bg = this_mean;
+                    break
+                else
+                    prev_skew = this_skew;
+                end
+            end
```

Comment:
See above comment - if this loop runs until the end without line 707 triggering then `bg` will be left as the median of all values but `obj.ibg` will be set to the lowest finite value - but the rest of the code assumes that `bg` is the mean of `obj.ibg` (e.g. `this_mean`).

I don't know how likely this edge case is...